apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.ejs

```diff
@@ -179,9 +184,16 @@
             <% } else { %>
             <tr>
               <th>Question</th>
-              <th class="text-center">Submission status</th>
+              <% if (assessment.type === 'Exam') { %><th class="text-center">Submission status</th><% } %>
             <% } %>
-            <% if (has_auto_grading_question && assessment.allow_real_time_grading) { %>
+
+            <% if (assessment.type === 'Homework') { %>
```

Comment:
To make this all easier to understand and modify, I'd recommend mirroring the order and structure of conditionals between the `<thead>` and `<tbody>` sections as closely as possible (if it's indeed possible to do that in a sensible way - it feels like it should be, but I'm not 100% sure if this suggestion would pan out). That's the thing I'm currently struggling the most with when reviewing.

Reply:
tbody already has that. Note that other than the question zone titles and question label, all other columns are already behind a big if/else on the assessment type. I'll try to rearrange the thead to make it a bit easier to follow.

Reply:
Hopefully the changes in 27c8d95c8 should make it easier to follow?

Reply:
This is better, thanks. To be more explicit with my suggestion, when I say "mirror the order and structure of conditionals", I'm referring to the fact that the `<thead>` leads with `if (assessment.type === 'Homework')`, whereas `<tbody>` leads with `if (assessment.type === 'Exam')`. IMO it'd be easier to mentally parse this page if e.g. both used `if (assessment.type === 'Homework') { ... } else { ... }`. Another difference: `<thead>` uses two `if (assessment.type === 'Homework') { ... } else { ... }` blocks back-to-back, whereas `<tbody>` uses just a single one, which makes it easier to parse what's happening in each case.

Reply:
I can rearrange the order. As for the two blocks back-to-back, they refer to similar functionality, so I'd prefer to keep them separate if you don't mind.

Reply:
Reordered in 5ebbebdc2.

Reply:
Thanks, I think I see now why the two sections can't perfectly mirror each other. It's because the header actually has multiple `<tr>` elements with `colspan` affecting the layout, whereas the body just has columns straight across, right?

Reply:
Yes, that's correct. It just so happens (by coincidence) that the cell that uses colspan has 3 columns on both exams and homeworks, but the 3 columns are different, so keeping them separate probably makes sense to help with that case.